inc/class-sentry.php

```diff
@@ -0,0 +1,159 @@
+
+	/**
+	 * Sentry DSN
+	 *
+	 * @var string
+	 */
+	protected static $dsn;
+
+	/**
+	 * @var \WP_User
+	 */
+	private $user;
+
+	/**
+	 * @since 5.5.3
+	 *
+	 * @return Sentry
+	 */
+	static public function init() {
+
+		if ( self::areEnvironmentAvailable() ) {
+			self::$dsn = env( 'SENTRY_DSN' );
+		}
+
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+			self::$instance->getCurrentUserForTracking();
+			if (
+				! is_null( env( 'SENTRY_INITIALIZE_PHP' ) ) &&
+				intval( env( 'SENTRY_INITIALIZE_PHP' ) ) === 1
+			) {
+				self::$instance->phpObserver();
+			}
+			if (
+				! is_null( env( 'SENTRY_INITIALIZE_JAVASCRIPT' ) ) &&
+				intval( env( 'SENTRY_INITIALIZE_JAVASCRIPT' ) ) === 1
+			) {
+				self::$instance->javascriptObserver();
+			}
+		}
+
+		return self::$instance;
+
+	}
+
+	/**
+	 * Get current user for tracking (if available)
+	 *
+	 * @return false|\WP_User
+	 */
+	public function getCurrentUserForTracking() {
+		$user_id = get_current_user_id();
+		$this->user = $user_id > 0 ? $user = get_userdata( $user_id ) : false;
+		return $this->user;
+	}
+
+	/**
+	 * Set user for tracking
+	 *
+	 * @param \WP_User $user
+	 * @return \WP_User
+	 */
+	public function setUserForTracking( \WP_User $user ) {
+		$this->user = $user;
+		return $this->user;
+	}
+
+	/**
+	 * Check if required environment variables are available to initialize Sentry's SDKs
+	 *
+	 * @return bool
+	 */
+	static private function areEnvironmentAvailable() {
+		return ! is_null( env( 'WP_ENV' ) ) && ! is_null( env( 'SENTRY_DSN' ) );
+	}
+
+
+	/**
+	 * Initialize Sentry for PHP using Sentry's PHP SDK
+	 *
+	 * @return bool
+	 */
+	public function phpObserver() {
+		try {
+			\Sentry\init(
+				[
+					'dsn' => self::$dsn,
+					'environment' => env( 'WP_ENV' ) ?: self::DEFAULT_ENVIRNOMENT,
+				]
+			);
+			if ( $this->user ) {
+				$user = $this->user;
+				\Sentry\configureScope(
+					function ( \Sentry\State\Scope $scope ) use ( $user ) {
+						$scope->setUser( [
+							'username' => $user->user_login,
+							'email' => $user->user_email,
+						] );
+					}
+				);
+			}
+			return true;
+		} catch ( \Exception $exception ) {
+			debug_error_log( 'Error initializing Sentry for PHP: ' . $exception->getMessage() );
+		}
+		return false;
+	}
+
+	/**
+	 * Get HTML JS tag inclusion for Sentry's JavaScript SDK
+	 *
+	 * @return string
+	 */
+	public function javascriptObserver() {
+		try {
+			$assets = new Assets( 'pressbooks', 'plugin' );
+			$src = $assets->getPath( 'scripts/sentry.js' );
+			wp_enqueue_script( self::WP_SCRIPT_NAME, $src );
+			$script_params = [
+				'dsn' => self::$dsn,
+				'environment' => env( 'WP_ENV' ) ?: self::DEFAULT_ENVIRNOMENT,
+				'user' => false,
+			];
+			if ( $this->user ) {
+				$script_params['user'] = [
+					'username' => $this->user->user_login,
+					'email' => $this->user->user_email,
+				];
+			}
+			wp_localize_script( self::WP_SCRIPT_NAME, 'SentryParams', $script_params );
```

Comment:
What's the purpose of this?

Reply:
It is used for pass variables to the JavaScript script from server side before create and compile the script. More references here:
- https://wpshout.com/building-a-magical-golden-bridge-from-php-to-javascript-with-wp_localize_script/
- https://medium.com/vunamhung/passing-php-variables-to-javascript-with-wp-localize-script-d99e165c1f9c